wallabagger/js/popup.js

```diff
@@ -409,26 +386,29 @@ PopupController.prototype = {
         chipClose.addEventListener('click',this.deleteTag.bind(this));
         return element.firstChild;
                 },
-
+    
     createTagChipNoClose: function(tagid,taglabel) {
         let element = document.createElement('div');
         element.innerHTML =`<div class="chip-sm" data-tagid="${tagid}" data-taglabel="${taglabel}"" style="cursor: pointer;"><span class="chip-name">${taglabel}</span></div>`;
         element.firstChild.addEventListener('click',this.onFoundTagChipClick.bind(this));
         return element.firstChild;        
                 },
-
+    
     clearTagInput:  function(){
          let tagsA = Array.prototype.slice.call(this.tagsInputContainer.childNodes);
          return tagsA.filter(e=> ( e.classList != null ) && e.classList.contains("chip-sm") )
                     .map( e=> { this.tagsInputContainer.removeChild(e); return 0; } );
     },
     
     createTags: function(data){
+
         this.articleTags = data;
+
+        this.dirtyTags = this.dirtyTags.filter( tag => this.articleTags.filter(atag => atag.slug === tag.slug).length === 0 );
+
         this.clearTagInput();
-        return this.articleTags.map( tag => {
+        this.articleTags.concat(this.dirtyTags).map( tag => {
```

Comment:
This is nessessary: when you add several tag quickly there are several requests and if we create only clean tag from response - all other dirty tag disappears and appears again only when their responses come. This flickering is annoyng. Tags added with this response clears from dirty at line 407